plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -932,7 +935,8 @@ void producer_plugin::resume() {
    //
    if (my->_pending_block_mode == pending_block_mode::speculating) {
       chain::controller& chain = my->chain_plug->chain();
-      chain.abort_block();
+      vector<transaction_metadata_ptr> aborted_trxs = chain.abort_block();
+      my->_unapplied_transactions.add_aborted( std::move( aborted_trxs ) );
```

Comment:
Why not just do `my->_unapplied_transactions.add_aborted( chain.abort_block() );`?

Reply:
Done